### PR TITLE
Dev/0.1

### PR DESCRIPTION
**Pull request by eyworldwide**
First PR.

### CODE REVIEW

File: src/LottieLoader.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  AssetPromise,
+  AssetType,
+  LoadItem,
+  ResourceManager,
+  Texture2D
+} from "oasis-engine";
+
+import { LottieResource, TypeRes } from './LottieResource';
+import { LottieRenderer } from './LottieRenderer';
+import { Entity } from "oasis-engine";
```
Comment (singlecoder) [inline src/LottieLoader.ts:13]: Should import Entity on the top

File: src/LottieRenderer.ts [hunk 2]
```diff
@@ -0,0 +1,327 @@
+import {
+	CompLottieLayer,
+	SpriteLottieLayer,
+	Tools
+} from './core';
+import { MeshBatcher } from './MeshBatcher';
+import { Matrix, Quaternion, Texture2D, Vector3 } from "oasis-engine";
```
Comment (singlecoder) [inline src/LottieRenderer.ts:7]: Should import together

File: src/LottieRenderer.ts [hunk 3]
```diff
@@ -0,0 +1,327 @@
...
+	// Temp variables for better performance
+	private tempPosition: Vector3 = new Vector3();
+	private tempTranslation: Vector3 = new Vector3();
+	private tempRotation: Quaternion = new Quaternion();
+	private tempScale: Vector3 = new Vector3();
+	private tempWorldMatrix: Matrix = new Matrix();
+	private tempLocalMatrix: Matrix = new Matrix();
+	private tempParentWorldMatrix: Matrix = new Matrix();
+
+	root: CompLottieLayer = null;
```
Comment (singlecoder) [inline src/LottieRenderer.ts:39]: Put the root property before private property

File: src/LottieRenderer.ts [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+
+	set res(value) {
+		this.resource = value;
+
+		const { width, height } = value;
+
+		this.width = width;
+		this.height = height;
+
```
Comment (singlecoder) [inline src/LottieRenderer.ts:48]: this.width = value.width;
this.height = value.height;

File: src/LottieRenderer.ts [hunk 5]
```diff
@@ -0,0 +1,327 @@
...
+			}
+		}
+
+		return layersMap;
+	}
+
+	private updateLayers(layers) {
+		const { vertices } = this.batch;
+
+		for (let i = 0; i < layers.length; i++) {
```
Comment (singlecoder) [inline src/LottieRenderer.ts:111]: like for (let i = 0, l = layers.length; i < l; i++) should be better

File: src/LottieRenderer.ts [hunk 6]
```diff
@@ -0,0 +1,327 @@
...
+		}
+
+		batch.setBufferData();
+
+		batch.getMaterial().shaderData.setTexture('map', this.resource.texture);
+
+		return batch;
+	}
+
+	private updateBuffer(layer, i, vertices) {
```
Comment (singlecoder) [inline src/LottieRenderer.ts:141]: _updateBuffer？

File: src/LottieLoader.ts [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+
+// @ts-ignore
```
Comment (GuoLei1990) [inline src/LottieLoader.ts:15]: Why have  `@ts-ignore`

File: src/LottieRenderer.ts [hunk 8]
```diff
@@ -0,0 +1,327 @@
...
+
+	repeats = 0;
+	infinite: boolean = false;
+	alternate: boolean = false;
+	direction: number = 1;
+	timeScale: number = 1;
+
+	private isPlaying: boolean = false;
+	private frame: number = 0;
+	private layers;
```
Comment (GuoLei1990) [inline src/LottieRenderer.ts:24]: add type

File: src/LottieRenderer.ts [hunk 9]
```diff
@@ -0,0 +1,327 @@
...
+	Tools
+} from './core';
+import { MeshBatcher } from './MeshBatcher';
+import { Matrix, Quaternion, Texture2D, Vector3 } from "oasis-engine";
+import { Script } from "oasis-engine";
+import { LottieResource } from './LottieResource';
+
+export { LottieLoader } from './LottieLoader';
+
+export class LottieRenderer extends Script {
```
Comment (GuoLei1990) [inline src/LottieRenderer.ts:13]: You can use the built-in `MeshRenderer` and a custom script `LottieAnimation` which extends `Script`, a little like `MeshRender`+ `Animator`.

In fact the 'LottieRenderer' you implement is just a lottie animation controller, so LottieAnimation is better!

**Comment by GuoLei1990** [comment]
Should follow development specifications: https://yuque.antfin-inc.com/oasisgroup/engine_dev/fnvozr

File: src/LottieRenderer.ts [hunk 10]
```diff
@@ -0,0 +1,327 @@
...
+	static unitsPerPixel: number = 1 / 128;
+
+	repeats = 0;
```
Comment (GuoLei1990) [inline src/LottieRenderer.ts:16]: if is public, you should add comments, if internal should  add "_" and `@internal`, if private should add "_" and `private`.

File: .aci.yml [hunk 11]
```diff
@@ -0,0 +1,40 @@
+# 不要修改该文件，会自动生成，详见 https://gitlab.alibaba-inc.com/node/ci
```
Comment (GuoLei1990) [inline .aci.yml:1]: This file should ignore.

**Pull request merged**